[Feature] 사용자 위치 권한 거부 시 에러 모달 처리 추가 및 컴포넌트 분리

yejiis opened the pull request:
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
<!---- Resolves: #(99) -->

## 💡 PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 파일 혹은 폴더명 수정 및 삭제
- [x] 기타 사소한 변경

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 변경 사항에 대한 테스트를 지켰는가
- [x] 커밋 메시지가 팀 컨벤션을 지켰는가
- [x] 브랜치 명이 규칙에 맞는가
- [x] 기능 정상 동작 확인 완료

##  📝 작업 내용
<!-- 작업한 내용을 작성해주세요 -->
- 위치 권한 거부 시 에러 모달 추가 (header, map 등 위치권한이 필요한 서비스에 사용되므로 전역적으로 상태관리하여 getCurrentCoordsUtil에서 통합관리할 수 있도록 설정함.)
- cluster 컴포넌트화 
- map 범위 조정 훅 추가

yejiis committed:
refactor: map-page 상태관리 수정 #99

yejiis committed:
feat: 사용자 위치 권한 거부 시 에러 모달 처리 추가 및 컴포넌트 분리 #99

yunchyn reviewed (dismissed):
1. LocationContent에 공백{' '}이 들어가 있습니다. 
2. MainMap alert에서  confirm 동작 onclick 함수가 따로 없으면 ShowCancleButton은 필요없지 않을까요?(확인이나 취소나 같은 동작을 한다면)

yejiis committed:
refactor: 불필요한 코드 제거 및 지도 중심 설정 함수 분리
- 공백 제거, 위치 권한 허용 모달 cancelButton 제거 및 컴포넌트화
- 지도 중심 설정 분리:  초기 맵 페이지 진입 시 params에 따른 조건 설정(initMap) 및 현재 위치(centerMapToCurrentLocation)으로 분리하여 관리 #99

yunchyn reviewed (approved):
확인했습니다.
